### DISCUSSION TITLE
Source code Build pipeline failure

### DISCUSSION THREAD

**Discussion by Munishdev97** [Sunbird-Ed/Community#644] [Installation]
In build source code pipelines 3 pipelines failure out of 66. We have checked with error logs, tried with make some changes in configuration. Still We are not able to solve the issues

**Comment by Munishdev97** [comment]
The following pipeline errors which we are facing in sunbirdED platform installation
 
ERRORS:

1. Build/Core/EncService : 

ERROR: failed to solve: process "/bin/sh -c apt update     && apt install -y zip python make g++     && npm i     && apt remove --purge -y python make g++     && apt-get autoremove -y     && rm -rf /var/lib/apt/lists/*" did not complete successfully: exit code: 100

ERROR: script returned exit code 1
Finished: FAILURE

2. Build/Core/Print:

ERROR: failed to solve: process "/bin/sh -c apt-get clean     && wget -q -O - https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add -     && sh -c 'echo \"deb [arch=amd64] http://dl.google.com/linux/chrome/deb/ stable main\" >> /etc/apt/sources.list.d/google.list'     && apt update && apt install fonts-indic -y     && apt-get install -y google-chrome-unstable gconf-service libasound2 libatk1.0-0 libc6 libcairo2 libcups2 libdbus-1-3 libexpat1 libfontconfig1 libgcc1 libgconf-2-4 libgdk-pixbuf2.0-0 libglib2.0-0 libgtk-3-0 libnspr4 libpango-1.0-0 libpangocairo-1.0-0 libstdc++6 libx11-6 libx11-xcb1 libxcb1 libxcomposite1 libxcursor1 libxdamage1 libxext6 libxfixes3 libxi6 libxrandr2 libxrender1 libxss1 libxtst6 ca-certificates fonts-liberation libappindicator1 libnss3 lsb-release xdg-utils wget       --no-install-recommends     && rm -rf /var/lib/apt/lists/*" did not complete successfully: exit code: 100

3. Build/Plugins/CollectionEditor:

npm WARN optional SKIPPING OPTIONAL DEPENDENCY: strip-ansi-cjs@npm:strip-ansi@^6.0.1 (node_modules/@isaacs/cliui/node_modules/strip-ansi-cjs):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Unsupported URL Type: npm:strip-ansi@^6.0.1
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: wrap-ansi-cjs@npm:wrap-ansi@^7.0.0 (node_modules/@isaacs/cliui/node_modules/wrap-ansi-cjs):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Unsupported URL Type: npm:wrap-ansi@^7.0.0
npm ERR! Linux 5.4.0-1109-azure
npm ERR! argv "/usr/local/lib/node-v6.17.1-linux-x64/bin/node" "/usr/bin/npm" "install"
npm ERR! node v6.17.1
npm ERR! npm  v3.10.10

npm ERR! Unsupported URL Type: npm:string-width@^4.2.0
npm ERR! 
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:

**Reply by gandham-santhosh** [reply]
Which version of Ed you are trying to install? 
@Munishdev97 Enc and print service were having issues with the base docker image which is updated. Please rebuild the print service and check. For Enc service please use release-5.1.0 branch to build. Collection Editors build and deploy jobs are not required as these are imported as npm package now. These details are updated in the 6.0 installation documentation

**Comment by Munishdev97** [comment]
Hello @gandham-santhosh,
 
1.For Build/core/Print, we tried rerunning the pipeline with given tag in documentation release-3.8.0_RC1 (in which base image used is node:8.11-slim) but it resulted in same error. So we tested with build parameter  release-3.8.0 branch(in which base image used is node:8-buster-slim) and it is working fine.

2.For Build/core/EncService, we used the branch tag as release-5.1.0 as updated in the new documentation but is failing with same error(Docker build step failing) as before.  We forked the repo and tried with base image as node:8-buster-slim and after that it is getting success similar to print pipeline.
Please review the PR here: https://github.com/project-sunbird/enc-service/pull/34

**Reply by gandham-santhosh** [reply]
@Munishdev97  print service tag is updated in the documentation with release-3.8.0_RC2  and Enc service PR is merged. Thanks
